homeassistant/components/mqtt/mixins.py

```diff
@@ -237,6 +259,25 @@ async def __call__(
         """Define setup_entities type."""
 
 
+async def async_get_platform_config_from_yaml(
+    hass: HomeAssistant, domain: str, schema: vol.Schema
+) -> list[ConfigType]:
+    """Return a list of validated configurations for the platform read from configuration.yaml."""
+
+    config_yaml: ConfigType = hass.data.get(DATA_MQTT_CONFIG, {})
+    if not (platform_configs := config_yaml.get(domain)):
+        return []
+    errors = []
+    try:
+        config = [schema(config) for config in platform_configs]
+    except (ValueError, vol.MultipleInvalid) as exc:
+        errors.append(exc)
+    if errors:
+        raise vol.MultipleInvalid(errors)
```

Comment:
What happens when we raise here, if we for example setup 2 fans, 1 good and 1 broken?
What if we setup 1 bad fan and 1 good light?

Reply:
The platform with broken entities will not be setup, since the loop is interrupted. We should return the not broken configs and log for the broken one's. But in that case we cannot raise here.

Reply:
Fixed. Now uses async_log_exception to log the exception and still returns valid configuration items.